pkg/resourcemanager/controller/managedresource/merger.go

```diff
@@ -145,14 +149,10 @@ func mergePodTemplate(oldPod, newPod *corev1.PodTemplateSpec, preserveResources
 	}
 
 	// Do not overwrite a PodTemplate's resource requests / limits if it is scaled by an HVPA
-	for _, newContainer := range newPod.Spec.Containers {
-		newContainerName := newContainer.Name
-
-		for _, oldContainer := range oldPod.Spec.Containers {
-			oldContainerName := oldContainer.Name
-
-			if newContainerName == oldContainerName {
-				mergeContainer(&oldContainer, &newContainer, preserveResources)
+	for i, newContainer := range newPod.Spec.Containers {
+		for j, oldContainer := range oldPod.Spec.Containers {
+			if newContainer.Name == oldContainer.Name {
+				mergeContainer(&oldPod.Spec.Containers[j], &newPod.Spec.Containers[i], preserveResources)
 			}
```

Comment:
Probably we can add `break` as container's name are unique in a pod and once we get the match no need check for the rest.
```suggestion
				mergeContainer(&oldPod.Spec.Containers[j], &newPod.Spec.Containers[i], preserveResources)
				break
			}
```

Reply:
That's a good point/observation, thank you! Done in ce8bede91.